audioipc/src/messages.rs

```diff
@@ -277,6 +282,49 @@ impl AssocRawFd for ClientMessage {
     }
 }
 
+#[cfg(windows)]
+use std::os::windows::io::RawHandle;
+
+#[cfg(windows)]
+pub trait AssocRawFd {
+    fn fd(&self) -> Option<([RawHandle; 3], u32)> {
+        None
+    }
+    fn take_fd<F>(&mut self, _: F)
+    where
+        F: FnOnce() -> Option<[RawHandle; 3]>,
+    {
+    }
+}
+
+#[cfg(windows)]
+impl AssocRawFd for ServerMessage {}
+
+#[cfg(windows)]
+impl AssocRawFd for ClientMessage {
+    fn fd(&self) -> Option<([RawHandle; 3], u32)> {
+        match *self {
+            ClientMessage::StreamCreated(ref data) => Some(([data.fds[0].as_raw(),
+                                                             data.fds[1].as_raw(),
+                                                             data.fds[2].as_raw()],
+                                                            data.target_pid)),
+            _ => None,
+        }
+    }
+
+    fn take_fd<F>(&mut self, f: F)
+    where
+        F: FnOnce() -> Option<[RawHandle; 3]>,
+    {
+        if let ClientMessage::StreamCreated(ref mut data) = *self {
+            let fds = f().unwrap();
```

Comment:
It's only unwrapped in the specific case where we're handling a StreamCreated, since we *must* have fds available.  In other cases (e.g. other impls of AssocRawFd or for other message types), it can be None.  I've replaced unwrap with expect to make it slightly clearer.